packages/apps/human-app/frontend/src/shared/contexts/color-mode/color-mode-handlers.ts

```diff
@@ -0,0 +1,32 @@
+import { hasColorMode } from './color-mode-settings';
+
+export const handleColorModeChange = (
+  matches: boolean,
+  setIsDarkMode: (value: boolean) => void
+) => {
+  if (hasColorMode()) {
+    return;
+  }
+  setIsDarkMode(matches);
+  if (matches) {
+    document.body.classList.add('dark-mode');
+  } else {
+    document.body.classList.remove('dark-mode');
+  }
+};
+
+export const runColorMode = (
+  fn: (matches: boolean) => void
```

Comment:
Tbh, the naming in this function is confusing, I would suggest to change it to
```
export const addColorSchemePrefsListener = (
  onChange: (prefersDark: boolean) => void
): (() => void) => {
  const query = window.matchMedia('(prefers-color-scheme: dark)');
  onChange(query.matches);

  const listener = (event: MediaQueryListEvent) => {
    onChange(event.matches);
  };
  query.addEventListener('change', listener);

  return () => {
    query.removeEventListener('change', listener);
  };
};
```

also it always returns `(() => void)`, no need in `undefined`